dev/com.ibm.ws.ejbcontainer.remote/src/com/ibm/ws/ejbcontainer/remote/internal/EJBRemoteRuntimeImpl.java

```diff
@@ -249,17 +250,23 @@ public synchronized void unbindAll(Object bindingDataObject) {
             }
         }
 
-        for (int i = bindingData.contextNames.length - 1; i >= 0 && !hasBindings(bindingData.contexts[i]); i--) {
+        // For contextNames, if completely empty of bindings, unbind the naming context from its parent context.
+        for (int i = bindingData.contextNames.length - 1; i >= 0 && isBindingContextEmpty(bindingData.contexts[i]); i--) {
```

Comment:
Seems we are starting at a leaf node, and if that context is empty, we unbind it from the parent, then back up one and check if that context is null, etc... all the way back to the root.  As soon as we find one that is not empty, we stop.

However, now we will stop as soon as we find a "null" context.... but what if the parent context exists and is empty, shouldn't we back up one and check the parent as well?  If it does exist and is empty, seems we should unbind it.